contracts/src/v0.8/dev/VRFCoordinatorV2.sol

```diff
@@ -0,0 +1,527 @@
+    }
+    fp.sender = sender;
+
+    bytes32 blockHash = blockhash(blockNum);
+    if (blockHash == bytes32(0)) {
+      blockHash = BLOCKHASH_STORE.getBlockhash(blockNum);
+      if (blockHash == bytes32(0)) {
+        revert BlockHashNotInStore(blockNum);
+      }
+    }
+    // The seed actually used by the VRF machinery, mixing in the blockhash
+    uint256 actualSeed = uint256(keccak256(abi.encodePacked(preSeed, blockHash)));
+    // solhint-disable-next-line no-inline-assembly
+    assembly { // Construct the actual proof from the remains of proof
+      mstore(add(proof, PRESEED_OFFSET), actualSeed)
+      mstore(proof, PROOF_LENGTH)
+    }
+    randomness = VRF.randomValueFromVRFProof(proof); // Reverts on failure
+  }
+
+  function getFeedData()
+    private
+    view
+    returns (
+        uint256
+    )
+  {
+    uint32 stalenessSeconds = s_config.stalenessSeconds;
+    bool staleFallback = stalenessSeconds > 0;
+    uint256 timestamp;
+    int256 linkEth;
+    (,linkEth,,timestamp,) = LINK_ETH_FEED.latestRoundData();
+    if (staleFallback && stalenessSeconds < block.timestamp - timestamp) {
+      linkEth = s_fallbackLinkPrice;
+    }
+    return uint256(linkEth);
+  }
+
+  function withdraw(
+    address recipient,
+    uint96 amount
+  )
+    external
+  {
+    if (s_withdrawableTokens[msg.sender] < amount) {
+      revert InsufficientBalance();
+    }
+    s_withdrawableTokens[msg.sender] -= amount;
+    assert(LINK.transfer(recipient, amount));
+  }
+
+  function getSubscription(
+    uint64 subId
+  )
+    external
+    view
+    returns (
+      Subscription memory
+    )
+  {
+    return s_subscriptions[subId];
+  }
+
+  function createSubscription(
+    address[] memory consumers // permitted consumers of the subscription
+  )
+    external
+    validConsumers(consumers)
+    returns (
+      uint64
+    )
+  {
+    currentSubId++;
+    s_subscriptions[currentSubId] = Subscription({
+      owner: msg.sender,
+      consumers: consumers,
+      balance: 0
+    });
+    emit SubscriptionCreated(currentSubId, msg.sender, consumers);
+    return currentSubId;
+  }
+
+  function updateSubscription(
+    uint64 subId,
+    address[] memory consumers // permitted consumers of the subscription
+  )
+    external
+    onlySubOwner(subId)
+    validConsumers(consumers)
+  {
+    address[] memory oldConsumers = s_subscriptions[subId].consumers;
+    s_subscriptions[subId].consumers = consumers;
+    emit SubscriptionConsumersUpdated(subId, oldConsumers, consumers);
+  }
+
+  function fundSubscription(
+    uint64 subId,
+    uint96 amount
+  )
+    external
+    onlySubOwner(subId)
+  {
+    if (s_subscriptions[subId].owner == address(0))  {
+      revert InvalidSubscription();
+    }
+    uint256 oldBalance = s_subscriptions[subId].balance;
+    s_subscriptions[subId].balance += amount;
+    LINK.transferFrom(msg.sender, address(this), amount);
+    emit SubscriptionFundsAdded(subId, oldBalance, s_subscriptions[subId].balance);
```

Comment:
Would it make sense to avoid this extra `s_subscriptions[subId].balance` read?

Reply:
After EIP2929 this is considered a "warm" read and so only costs 100 gas, but nevertheless makes sense to just make it oldBalance + amount